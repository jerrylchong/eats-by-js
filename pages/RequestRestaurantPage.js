import React, { useState, useEffect } from 'react';
import {
    StyleSheet,
    TouchableOpacity,
    SafeAreaView,
    TextInput,
    KeyboardAvoidingView,
    Text,
    View,
    Platform,
    ImageBackground,
    Dimensions, BackHandler
} from "react-native";
import BackButton from "../component/BackButton";
import {useSafeArea} from "react-native-safe-area-context";

function RequestRestaurantPage({ navigation }) {
    const [title, setTitle] = useState('');
    const [desc, setDesc] = useState('');
    const [location, setLocation] = useState('');
    const [operatingHours, setOperatingHours] = useState('');
    const [contact, setContact] = useState('');
    const [rating, setRating] = useState('');
    const [tags, setTags] = useState('');

    useEffect(() => {
        const backAction = () => {
            navigation.goBack();
            return true;
        };

        const backHandler = BackHandler.addEventListener(
            "hardwareBackPress",
            backAction
        );

        return () => backHandler.remove();
    }, []);

    const submit = () => {
        return (
            navigation.goBack()
        )
    }

    const insets = useSafeArea();

    return (
        <View style = {[
            styles.container,
            {paddingTop: insets.top, paddingLeft: insets.left, paddingRight: insets.right}
        ]}>
            <ImageBackground style = {styles.background} source={require('../assets/background.png')}/>
            <BackButton white={false} style={styles.back} onPress={navigation.goBack}/>
            <Text style = {styles.header}>Request a Restaurant</Text>
            <KeyboardAvoidingView behavior={Platform.OS == "ios" ? "padding" : "height"} style = {styles.list}>
                <TextInput
                    style={styles.input}
                    placeholder="Name"
                    onChangeText={(text) => {setTitle(text)}}
                    value={title}/>
                <TextInput
                    style={styles.input}
                    placeholder="Description"
                    onChangeText={(text) => {setDesc(text)}}
                    value={desc}/>
                <TextInput
                    style={styles.input}
                    placeholder="Location"
                    onChangeText={(text) => {setLocation(text)}}
                    value={location}/>
                <TextInput
                    style={styles.input}
                    placeholder="Operating Hours"
                    onChangeText={(text) => {setOperatingHours(text)}}
                    value={operatingHours}/>
                <TextInput
                    style={styles.input}
                    placeholder="Contact No."
                    onChangeText={(text) => {setContact(text)}}
                    value={contact}/>
                <TextInput
                    style={styles.input}
                    placeholder="Tags (each tag separated by a space)"
                    onChangeText={(text) => {setTags(text)}}
                    value={tags}/>
            </KeyboardAvoidingView>
            <View style = {styles.buttonShadow}>
                <TouchableOpacity style = {styles.button} onPress = {submit}>
                    <Text style = {styles.buttonText}>Request</Text>
                </TouchableOpacity>
            </View>
        </View>
    )
}

export default RequestRestaurantPage

const windowWidth = Dimensions.get('window').width;

const styles = StyleSheet.create({
    background: {
        position:'absolute',
        bottom: 0,
        width: '100%',
        height: Dimensions.get('window').width * 728/1668,
    },
    container: {
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#fff',
        alignItems: 'center',
        minHeight: Math.round(Dimensions.get('window').height),
    },
    back: {
        position: 'absolute',
        top: Platform.OS == "ios" ? '5%' :'2%',
        left: '2%'
    },
    header: {
        position: 'relative',
        top: '5%',
        fontSize: 20,
        marginTop: '5%',
        fontFamily: 'Ubuntu-Medium'
    },
    list: {
        position: 'relative',
        top: '6%',
        width: '80%',
        height: '50%',
        alignItems: 'center',
        marginTop: '5%',
        justifyContent: 'space-between'
    },
    inputHeader: {
        flexDirection: 'row',
        alignSelf: 'flex-start'
    },
    input: {
        borderRadius: windowWidth * 0.35,
        backgroundColor: '#d9d9d9',
        fontSize: 12,
        width: windowWidth * 0.8,
        height: windowWidth * 0.11,
        paddingHorizontal: '5%',
        fontFamily: 'Ubuntu',
        color: '#404040'
    },
    buttonShadow: {
        position: 'relative',
        top: '20%',
        width: Dimensions.get('window').width * 0.25,
        height: Dimensions.get('window').width * 0.08,
        backgroundColor: '#ff6961',
        borderRadius: 20,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
    },
    button: {
        height: '100%',
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        paddingHorizontal: 20,
        paddingVertical: 3,
    },
    buttonText: {
        fontSize: 12,
        color: 'white',
        fontFamily: 'Ubuntu'
    }
})