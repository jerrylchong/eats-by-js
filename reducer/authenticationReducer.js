const authenticationReducer = (state = "not_set", action) => {
    switch (action.type) {
        case 'SET_TOKEN':
            return action.payload;
        case 'UNSET_TOKEN':
            return "not_set";
        default:
            return state;
    }
};
export default authenticationReducer
