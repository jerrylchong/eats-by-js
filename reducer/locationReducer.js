const locationReducer = (state = {hasLocation : false, coords : {}}, action) => {
    switch (action.type) {
        case 'SET_LOCATION':
            return {
                hasLocation : true,
                coords : {lat: action.payload.coords.latitude, lng: action.payload.coords.longitude}
            }
        case 'UNSET_LOCATION':
            return {
                hasLocation : false,
                coords : {lat: null, lng: null}
            }

        default:
            return state;
    }
};
export default locationReducer;
