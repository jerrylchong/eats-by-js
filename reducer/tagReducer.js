const tagReducer = (state = {tagData: [], isEmpty: true}, action) => {
    switch (action.type) {
        case 'SET_TAGS':
            return {
                tagData : action.payload,
                isEmpty: false
            }
        case 'UNSET_TAGS':
            return {
                tagData : [],
                isEmpty: true
            }

        default:
            return state;
    }
};
export default tagReducer;
