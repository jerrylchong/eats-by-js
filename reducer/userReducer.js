const userReducer = (state = {isLoggedIn: false, user_data:{}}, action) => {
    switch (action.type) {
        case 'SET_USER':
            return {
                isLoggedIn : true,
                user_data : action.payload
            }
        case 'UNSET_USER':
            return {
                isLoggedIn : false,
                user_data : {}
            }

        default:
            return state;
    }
};
export default userReducer;
