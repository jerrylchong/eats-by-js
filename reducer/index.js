import {combineReducers} from "redux";
import authenticationReducer from "./authenticationReducer"
import userReducer from "./userReducer";
import locationReducer from "./locationReducer";
import tagReducer from "./tagReducer";


export default combineReducers({
    token: authenticationReducer,
    user: userReducer,
    location: locationReducer,
    tags: tagReducer
})
