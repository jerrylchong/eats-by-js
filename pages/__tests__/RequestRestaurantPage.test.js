import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import RequestRestaurantPage from "../RequestRestaurantPage";
import {SafeAreaProvider} from "react-native-safe-area-context";
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import reducer from '../../reducer';

// snapshot test
test('renders correctly', () => {
    const store = createStore(reducer);
    const tree = renderer.create(
        <Provider store={store}>
            <SafeAreaProvider initialSafeAreaInsets={{ top: 0, left: 0, right: 0, bottom: 0 }}>
                <RequestRestaurantPage
                    navigation={{navigate: jest.fn()}}
                />
            </SafeAreaProvider>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });