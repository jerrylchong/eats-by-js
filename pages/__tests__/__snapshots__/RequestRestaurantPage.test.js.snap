// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "column",
          "minHeight": 1334,
        },
        Object {
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "bottom": 0,
          "height": 327.33812949640287,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 327.33812949640287,
              "width": "100%",
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "left": "2%",
          "position": "absolute",
          "top": "5%",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 13,
              "width": 15,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "fontFamily": "Ubuntu-Medium",
          "fontSize": 20,
          "marginTop": "5%",
          "position": "relative",
          "top": "5%",
        }
      }
    >
      Request a Store
    </Text>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": "50%",
            "justifyContent": "space-between",
            "marginTop": "5%",
            "position": "relative",
            "top": "6%",
            "width": "80%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Name"
        placeholderTextColor="#404040"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "borderRadius": 262.5,
            "color": "#404040",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontWeight": "normal",
            "height": 82.5,
            "paddingHorizontal": "5%",
            "width": 600,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Address"
        placeholderTextColor="#404040"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "borderRadius": 262.5,
            "color": "#404040",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontWeight": "normal",
            "height": 82.5,
            "paddingHorizontal": "5%",
            "width": 600,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 262.5,
              "flexDirection": "row",
              "height": 82.5,
              "paddingHorizontal": "5%",
              "width": 600,
            },
            Object {
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#404040",
              "fontFamily": "Ubuntu",
              "fontSize": 12,
            }
          }
        >
          Location:
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#ff6961",
                "borderRadius": 10,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Ubuntu-Bold",
                  "fontSize": 10,
                  "paddingHorizontal": 9,
                  "paddingVertical": 3,
                }
              }
            >
              Manual Input
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#ff6961",
                "borderRadius": 10,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Ubuntu-Bold",
                  "fontSize": 10,
                  "paddingHorizontal": 9,
                  "paddingVertical": 3,
                }
              }
            >
              Use Current Location
            </Text>
          </View>
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Operating Hours"
        placeholderTextColor="#404040"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "borderRadius": 262.5,
            "color": "#404040",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontWeight": "normal",
            "height": 82.5,
            "paddingHorizontal": "5%",
            "width": 600,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Contact No."
        placeholderTextColor="#404040"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "borderRadius": 262.5,
            "color": "#404040",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontWeight": "normal",
            "height": 82.5,
            "paddingHorizontal": "5%",
            "width": 600,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Tags (each tag separated by a space)"
        placeholderTextColor="#404040"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "borderRadius": 262.5,
            "color": "#404040",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontWeight": "normal",
            "height": 82.5,
            "paddingHorizontal": "5%",
            "width": 600,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ff6961",
          "borderRadius": 20,
          "elevation": 5,
          "height": 60,
          "position": "relative",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "top": "20%",
          "width": 187.5,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 3,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Ubuntu",
              "fontSize": 12,
            }
          }
        >
          Request
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
